### PR TITLE
small changes

### LINKED ISSUE #9: Remove .dockerignore
We've moved all the protobufs into another repo, so we don't need `.dockerignore` anymore

### LINKED ISSUE #10: Add `-tags=nomsgpack` to build invocation in Dockerfile
We apparently do not use msgpack, so we can disable it to save some space.

### LINKED ISSUE #11: Remove proto compilation step from Taskfile.yaml

### LINKED ISSUE #15: Use Gin's release mode
As per Gin's output:
```
[GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
 - using env:	export GIN_MODE=release
 - using code:	gin.SetMode(gin.ReleaseMode)
```

We should set `GIN_MODE` env variable to `release` in Dockerfile

### PR DESCRIPTION
**Pull request by alexproskurov**
Closes #9
Closes #10 
Closes #11 
Closes #15

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,9 +11,8 @@ RUN go mod download
 
 COPY cmd cmd
 COPY internal internal
-COPY pkg pkg
 
-RUN go build -o /server ./cmd/main.go
+RUN go build -o /server ./cmd/main.go -tags=nomsgpac
```
Comment (knightpp) [inline Dockerfile]: The tag should be nomsgpac**k**. Per `go help build` go has the following argument format, so the build flags should be just before package/file names.
```
usage: go build [-o output] [build flags] [packages]
```

**Review by knightpp — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (alexproskurov) [reply Dockerfile]: my bad, fixed it.

File: Dockerfile [hunk 2]
```diff
@@ -10,9 +11,8 @@ RUN go mod download
 
 COPY cmd cmd
 COPY internal internal
-COPY pkg pkg
 
-RUN go build -o /server ./cmd/main.go
+RUN go build -o /server -tags=nomsgpac ./cmd/main.go
```
Comment (knightpp) [inline Dockerfile]: Please add `k` to the end of the tag name. It seems not right, because the tag should be `nomsgpack`

**Review by knightpp — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (alexproskurov) [reply Dockerfile]: (:

**Review by knightpp — APPROVED** [review]

**Pull request merged**